File: frontend/clubconnect/lib/insfrastructure/models/configEvento.dart
import 'dart:convert';

class ConfigEventos {
  String? id;
  DateTime fechaInicio;
  DateTime fechaFinal;
  String horaInicio;
  String horaFinal;
  String idEquipo;
  String descripcion;
  String lugar;
  int diaRepetible;
  String titulo;

  ConfigEventos({
    this.id,
    required this.fechaInicio,
    required this.fechaFinal,
    required this.horaInicio,
    required this.horaFinal,
    required this.idEquipo,
    required this.descripcion,
    required this.lugar,
    required this.diaRepetible,
    required this.titulo,
  });

  factory ConfigEventos.fromRawJson(String str) =>
      ConfigEventos.fromJson(json.decode(str));

  String toRawJson() => json.encode(toJson());

  factory ConfigEventos.fromJson(Map<String, dynamic> json) => ConfigEventos(
        id: json["id"],
        fechaInicio: DateTime.parse(json["fecha_inicio"]),
        fechaFinal: DateTime.parse(json["fecha_final"]),
        horaInicio: json["hora_inicio"],
        horaFinal: json["hora_final"],
        idEquipo: json["id_equipo"],
        descripcion: json["descripcion"],
        lugar: json["lugar"],
        diaRepetible: json["dia_repetible"],
        titulo: json["titulo"],
      );

  Map<String, dynamic> toJson() => {
        "id": id,
        "fecha_inicio": fechaInicio.toIso8601String(),
        "fecha_final": fechaFinal.toIso8601String(),
        "hora_inicio": horaInicio,
        "hora_final": horaFinal,
        "id_equipo": idEquipo,
        "descripcion": descripcion,
        "lugar": lugar,
        "dia_repetible": diaRepetible,
        "titulo": titulo,
      };
}

File: backend/routes/config_evento.js
const configEventoController = require("../controllers/configEventoController");
module.exports = (app) => {
  // Crear configuracion y eventos en la fecha dada
  app.post("/configEvento", async (req, res) => {
    const { fecha_inicio, fecha_final, hora_inicio, hora_final, id_equipo, descripcion, lugar, titulo, dia_repetible } = req.body;
    const response = await configEventoController.addConfigEvent(fecha_inicio, fecha_final, hora_inicio, hora_final, id_equipo, descripcion, lugar, titulo, dia_repetible);
    return response.statusCode === 400
      ? res.status(400).send({ message: response.message })
      : response.statusCode === 500
      ? res.status(500).send({ message: response.message })
      : res.status(201).send({
          data: response.data,
          message: response.message,
        });
  });
  // Otener la configracion de los eventos
  app.get("/configEventos", async (req, res) => {
    const { id_equipo } = req.query;
    const response = await configEventoController.getTeamConfigEvent(id_equipo);
    return response.statusCode === 400
      ? res.status(400).send({ message: response.message })
      : response.statusCode === 500
      ? res.status(500).send({ message: response.message })
      : res.status(200).send({
          data: response.data,
          message: response.message,
        });
  });
  // ELiminar configuracion de eventos recurrentes y eventos activos asociados
  app.delete("/configEvento", async (req, res) => {
    const { id_config } = req.query;
    const response = await configEventoController.deleteConfigEvent(id_config);
    return response.statusCode === 400
      ? res.status(400).send({ message: response.message })
      : response.statusCode === 500
      ? res.status(500).send({ message: response.message })
      : res.status(200).send({
          data: response.data,
          message: response.message,
        });
  });
  app.put("/configEvento", async (req, res) => {
    const { id, fecha_inicio, fecha_final, hora_inicio, hora_final, id_equipo, descripcion, lugar, titulo, dia_repetible } = req.body;
    const response = await configEventoController.editConfigEvento(id, fecha_inicio, fecha_final, hora_inicio, hora_final, id_equipo, descripcion, lugar, titulo, dia_repetible);
    return response.statusCode === 400
      ? res.status(400).send({ message: response.message })
      : response.statusCode === 500
      ? res.status(500).send({ message: response.message })
      : res.status(200).send({
          data: response.data,
          message: response.message,
        });
  });
};
